test/builders/build/eth-api/libraries/web3py/test_deploy_contract.py

```diff
@@ -0,0 +1,130 @@
+from turtle import bye
+from web3 import Web3
+import unittest
+import solcx 
+import os
+
+def compile_contract():
+    ROOT_DIR = os.path.abspath(os.curdir)
+    temp_file = solcx.compile_files([ROOT_DIR + '/contracts/Incrementer.sol'], output_values=["abi", "bin"])
+    abi = temp_file['contracts/Incrementer.sol:Incrementer']['abi']
+    bytecode = temp_file['contracts/Incrementer.sol:Incrementer']['bin']
+
+    return { "abi": abi, "bytecode": bytecode }
+
+def deploy_contract(abi, bytecode, web3, alice, alice_pk):
+    Incrementer = web3.eth.contract(abi=abi, bytecode=bytecode)
+
+    construct_txn = Incrementer.constructor(5).buildTransaction(
+        {
+            'from': alice,
+            'nonce': web3.eth.get_transaction_count(alice),
+        }
+    )
+    tx_create = web3.eth.account.sign_transaction(construct_txn, alice_pk)
+    tx_hash = web3.eth.send_raw_transaction(tx_create.rawTransaction)
+    tx_receipt = web3.eth.wait_for_transaction_receipt(tx_hash)
+
+    return tx_receipt
+
+class TestDeployContract(unittest.TestCase):
+    def setUp(self):
+        self.web3 = Web3(Web3.HTTPProvider('http://127.0.0.1:9933'))
+
+        # Use default account for Alice
+        self.alice = "0xf24FF3a9CF04c71Dbc94D0b566f7A27B94566cac"
+        self.alice_pk = "0x5fb92d6e98884f76de468fa3f6278f8807c48bebc13595d45af5bdc4da702133"
+
+    def test_compile_to_bytecode(self):
+        bytecode = compile_contract()["bytecode"]
+        self.assertNotEqual(bytecode, '')
```

Comment:
here I would compare the bytecode to whatever you expect it to be right? If you are using a fixed solidity compiler version it is deterministic

Reply:
Umm not sure. I guess we could leave it like this because we have to expect that the compiler is doing its job correctly? But we also should just ensure that our examples work, so if our examples are using a fixed solidity version we should stick to that

Reply:
Yeah I mean our examples aren't using a fixed solidity version, so I think if we set the versions to ^0.8.0 that would be the correct path forward, which means I wouldn't be able to check exact bytecode in the web3py test, but at least there is a workaround for the web3.js and ethers.js tests that checks the deployed bytecode